### DISCUSSION TITLE
Idea For Tutorial

### DISCUSSION THREAD

**Discussion by RPAlessio** [su2code/SU2#1490] [Q&A]
Dear all,

I am taking a course called Software Simulation Engineering and one of the course challenges is to make a contribution to an Open-Source Simulation Software. I chose SU2 and would like to contribute with a tutorial concerning a 2D internal compressible flow, possibly with heat transfer from the environment to the pipe wall. I noticed that there are no internal compressible flow tutorials in the website, and therefore I came up with this idea. I would like to know if this topic is of any interest to SU2. If so, I could develop it and create a pull request for its evaluation.

Looking forward to your feedback.

Best regards.

**Comment by pcarruscag** [comment]
Hi,
More tutorials are always good, what do you have in mind exactly when you say heat transfer? Do you want to model the pipe wall as a solid heat transfer domain? Or just as a normal heatflux boundary?

**Reply by RPAlessio** [reply]
Hi,

Thank you for such a quick reply.

Actually as a solid heat transfer wall with thermal resistance. The idea is to have an Environment Temperature T_inf = 1000K or 900K around the wall, and then heat is transferred by natural convection from the environment to the "outside" surface of the domain. The heat is then conducted through the pipe wall, which has a thermal resistance Rt, and then convected away by the compressible flow inside the pipe.

**Reply by pcarruscag** [reply]
Sounds like an interesting multiphysics problem. I guess you would use the incompressible solver for the natural convection part?
@TobiKattmann and @oleburghardt are the resident heat transfer experts.
We have developer meetings every Wednesday at 3pm UK (https://meet.jit.si/SU2_DevMeeting), you are welcome to join to talk more about this "in person" if you want.

We try to keep tutorials cases relatively small (maybe max 200k nodes) because we include them in the regression tests, were you thinking of doing this in 2D or 3D?

**Reply by bigfooted** [reply]
So if I understand correctly, the simulation will then have 3 zones:
1. compressible fluid flow inside the pipe
2. heat conduction through the solid pipe
3. incompressible air outside the pipe

**Reply by TobiKattmann** [reply]
Hi Alessio

A good start would be to use the Heat transfer/Convection boundary (see e.g. [this youtube video](https://www.youtube.com/watch?v=ygR31_vhvJI&t=915s)) with compressible or incompressible flow (see PR #1226 ) before actually modelling that part explicitly. 

@oleburghardt added CHT with compressible flow in PR #838 . You can look in the PR for the associated Testcases which demonstrate the use and might act as a starting point.

A tutorial would be very much appreciated, but in the timeframe of a university course without much SU2-experience make sure to not overload yourself. Shoot for a Minimum Viable Product first (e.g. 1 fluid zone just using a convection boundary) and get everything done (PR's open to Tutorials and the website with the necessary reviews where you will have to iterate a bit until it can be pushed). If you still have time left then you can still choose whether you go out and enjoy life or add more complexity to your case. A small contributions that is well executed is much more valuable in the long term than a big one that might not even get merged.

**Reply by RPAlessio** [reply]
Hi!

Alright. Thank you for the tip. I will start working on the tutorial with the mentioned simplified approach soon and hopefully it will work. Also, thank you for the links and references. I appreciate it.

**Reply by RPAlessio** [reply]
Hello to all,

I am developing the tutorial, but a made a small change. I considered incompressible instead of compressible flow. I hope that this is still of interest for the SU2 community.

**Reply by bigfooted** [reply]
Of course an incompressible setup is interesting as well. If you put your tutorial (mesh, config file, pictures) somewhere on github before submitting it, we could have a look at the setup and give some early feedback if you'd like.

**Reply by RPAlessio** [reply]
Alright, sounds good. Will do so.

**Reply by RPAlessio** [reply]
Hi @bigfooted,
I am attaching here `.txt` (representing the `.cfg` and `.su2` files)  and a `.md` (description of the tutorial) files along with pictures of the velocity field in x direction and the temperature field.
[tut.md](https://github.com/su2code/SU2/files/8026908/tut.md)
[lam_inc_int_convec.txt](https://github.com/su2code/SU2/files/8026937/lam_inc_int_convec.txt)
[sym.txt](https://github.com/su2code/SU2/files/8026923/sym.txt)
![vel_x_field](https://user-images.githubusercontent.com/93247180/153063571-ad2028d8-92ff-4e85-923d-a34db7370779.png)
![temperature_field](https://user-images.githubusercontent.com/93247180/153063537-befe14fe-7b0f-41a4-8612-ff2bbdb87278.png)

**Reply by bigfooted** [reply]
OK, so this is a planar laminar channel flow with a heated wall, right? Have you changed anything compared to the existing tutorial?
And you want to then add the solid wall and the ambient air outside of the wall? Have you already looked at the multizone tutorials for this setup?

**Reply by RPAlessio** [reply]
Not quite. It is a channel flow, yes, but what is heated is the flow inside it. Yes, I changed. I added the convective heat flux boundary condition and the symmetry boundary condition (flow is symmetric, so no need to have those extra elements) and the geometry is different. Yes, I checked that tutorial, but did not find it helpful for what I wanted to do. Plus, I followed the recommendations above in this discussion, where I should make it simple (especially concerning the convective heat flux bc), but complete and nicely explained.
